multimodal/src/autogluon/multimodal/utils/data.py

```diff
@@ -430,18 +430,13 @@ def data_to_df(
         data = pd.DataFrame(data)
     elif isinstance(data, list):
         assert len(data) > 0, f"Expected data to have length > 0, but got {data} of len {len(data)}"
-        if contains_valid_images(data):
-            data_dict = {"image": data}
-            data = pd.DataFrame(data_dict)
+        if header is None:
+            data = pd.DataFrame(data)
         else:
-            if header is None:
-                data = pd.DataFrame(data)
-            else:
-                data = pd.DataFrame({header: data})
+            data = pd.DataFrame({header: data})
     elif isinstance(data, str):
-        if contains_valid_images(data):
-            data_dict = {"image": [data]}
-            data = pd.DataFrame(data_dict)
+        if is_valid_imagepath(data):
+            data = pd.DataFrame({"image": data})
```

Comment:
`data` -> `[data]`

Reply:
BTW, do we need the header `image`? I guess no.

Reply:
Hey Zhiqiang. It seems like if we don't supply the header, we'll get a `KeyError`. 
`    def test_classification_str_list_input():
        download_dir = "./ag_automm_tutorial_imgcls"
        train_data, test_data = shopee_dataset(download_dir)

        model_path = f"./tmp/{uuid.uuid4().hex}-automm_shopee"
        predictor = MultiModalPredictor(label="label", path=model_path)
        predictor.fit(
            train_data=train_data,
            time_limit=30,  # seconds
        )  # you can trust the default config, e.g., we use a `swin_base_patch4_window7_224` model

        image_path = test_data.iloc[0]["image"]

        predictions_str = predictor.predict(image_path)
>       predictions_list1 = predictor.predict([image_path])

test_classification.py:21:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
../../../src/autogluon/multimodal/predictor.py:2383: in predict
    outputs = self._predict(
../../../src/autogluon/multimodal/predictor.py:2038: in _predict
    data, df_preprocessor, data_processors = self._on_predict_start(
../../../src/autogluon/multimodal/predictor.py:1875: in _on_predict_start
    if is_imagepath_column(data=data[col_name], col_name=col_name, sample_n=1):
/home/ubuntu/.local/lib/python3.9/site-packages/pandas/core/frame.py:3805: in __getitem__
    indexer = self.columns.get_loc(key)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = RangeIndex(start=0, stop=1, step=1), key = 'image', method = None, tolerance = None

    @doc(Int64Index.get_loc)
    def get_loc(self, key, method=None, tolerance=None):
        if method is None and tolerance is None:
            if is_integer(key) or (is_float(key) and key.is_integer()):
                new_key = int(key)
                try:
                    return self._range.index(new_key)
                except ValueError as err:
                    raise KeyError(key) from err
            self._check_indexing_error(key)
>           raise KeyError(key)
E           KeyError: 'image'

/home/ubuntu/.local/lib/python3.9/site-packages/pandas/core/indexes/range.py:395: KeyError`

Reply:
I guess that's because the `check_if_imagepath_and_add_column_title ` function. Could you try removing this function? I think it may be not required.